### PR TITLE
Add support for workers

### PR DESCRIPTION
**Pull request by mnagy**
Trello: https://trello.com/c/o6ijZ01e/892-5-update-python-image-to-autoconfigure-based-on-available-memory-evg

### CODE REVIEW

**Comment by mnagy** [comment]
@bparees PTAL?

**Comment by mnagy** [comment]
Hmm. Should I add the functionality to 3.5 as well?

**Comment by bparees** [comment]
@mnagy yes

File: 2.7/s2i/bin/run [hunk 1]
```diff
@@ -13,7 +15,21 @@ function should_migrate() {
...
+function get_default_web_concurrency() {
+  limit_vars=$(cgroup-limits)
+  local $limit_vars
+  if [ -z "${NUMBER_OF_CORES:-}" ]; then
+    echo 1
+    return
+  fi
+
+  local max=$((NUMBER_OF_CORES*4))
+  local default=$(((${MEMORY_LIMIT_IN_BYTES:-MAX_MEMORY_LIMIT_IN_BYTES}/1024/1024 - 40) / 3))
```
Comment (bparees) [inline 2.7/s2i/bin/run]: can you add a comment explaining what this equation represents?  it looks like 40 megs of overhead and 3 megs per worker?

Comment (bparees) [inline 2.7/s2i/bin/run]: my comment seems to have gotten lost, sorry if you get this twice, but:  please update this w/ a comment indicating what this calculation is doing (looks like it assumes 40megs of overhead, then 3 megs per worker?)

Comment (bparees) [inline 2.7/s2i/bin/run]: (and update it in all the versions of course)

**Comment by wewang58** [comment]
@mnagy the pr no python3.5 related

**Comment by mnagy** [comment]
@bparees addressed comments, added 3.5.

@wewang58 it is now. :)

**Comment by bparees** [comment]
lgtm

**Pull request merged**